libraries/config/src/main/java/org/dataportabilityproject/config/CommonSettings.java

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.io.InputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Settings used in both the API and worker servers.
+ *
+ * These settings are parsed from config/common.yaml and config/env/common.yaml.
+ *
+ * <p>The env/ file is an optional mechanism to allow environment-specific overrides of settings in
+ * the base common.yaml. This requires a distribution's build.gradle to copy the common.yaml file
+ * for the appropriate environment into config/env/common.yaml in the jar. TODO(#202): Do this for
+ * the sample Google deployments as a reference.
+ */
+public class CommonSettings {
+  public enum Environment {
+    LOCAL,
+    TEST,
+    QA,
+    PROD
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(CommonSettings.class);
+  /**
+   * We use a static singleton to facilitate:
+   * 1. core libraries injecting a CommonSettings (provided in {@code CommonSettingsModule})
+   * 2. extensions, which do not necessarily use Guice, retrieving the settings statically here
+   */
+  private static CommonSettings commonSettings;
+
+  private static final String COMMON_SETTINGS_PATH = "config/common.yaml";
+  private static final String ENV_COMMON_SETTINGS_PATH = "config/env/common.yaml";
+
+  private final String cloud;
+  private final Environment environment;
+
+  public CommonSettings() {
+    cloud = null;
+    environment = null;
+  }
+
+  @JsonCreator
+  public CommonSettings(
+      @JsonProperty(value = "cloud", required = true) String cloud,
+      @JsonProperty(value = "environment", required = true) Environment environment) {
+    this.cloud = cloud;
+    this.environment = environment;
+  }
+
+  public synchronized static CommonSettings getCommonSettings() {
+    if (commonSettings != null) {
+      return commonSettings;
+    }
+
+    // Any setting in both a base and env config will be overridden by the env definition.
+    // This is enforced by the ordering of the settings files in the list below.
+    try {
+      ImmutableList<String> settingsFiles = ImmutableList.<String>builder()
+          .add(COMMON_SETTINGS_PATH)
+          .add(ENV_COMMON_SETTINGS_PATH)
+          .build();
+      CommonSettings tempCommonSettings = getCommonSettings(settingsFiles);
+      logger.debug("Parsed flags: {}", tempCommonSettings);
+      commonSettings = tempCommonSettings;
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Problem parsing common settings", e);
+    }
+    return commonSettings;
+  }
+
+  public String getCloud() {
+    return cloud;
```

Comment:
do we need an intialized or null check in case this is called before getCommonSettings?

Reply:
this one isn't static so it would be called on an instance. I just removed the null-init ctor so I think it should be ok now.